flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpClient.java

```diff
@@ -113,8 +113,8 @@ public void send(DSeries request) throws Exception {
 		client.newCall(r).enqueue(EmptyCallback.getEmptyCallback());
 	}
 
-	public static String serialize(Object obj) throws JsonProcessingException {
-		return MAPPER.writeValueAsString(obj);
+	public static String serialize(Object obj, ObjectMapper mapper) throws JsonProcessingException {
+		return mapper.writeValueAsString(obj);
```

Comment:
I understand you changed this for the tests, but it's a bit ugly. Maybe add a comment?

Reply:
Strictly speaking this mapper is only necessary for having deterministic test results. The proper solution would probably be parsing the JSON response and compare the results not as a string.

Reply:
Yeah I wasn't too happy with it either; it is especially finicky since every setting the reporter uses for the mapper is ignored by the test :/